gordo_components/cli/cli.py

```diff
@@ -172,70 +104,50 @@ def build(
     model_parameter: List[Tuple]
         List of model key-values, wheres the values will be injected into the model
         config wherever there is a jinja variable with the key.
-
-    evaluation_config: dict
-        Dict of parameters which are exposed to ModelBuilder.build.
-            - cv_mode: str
-                String which enables three different modes, represented as a key value in evaluation_config:
-                * cross_val_only: Only perform cross validation
-                * build_only: Skip cross validation and only build the model
-                * full_build: Cross validation and full build of the model, default value
-                Example::
-
-                    {"cv_mode": "cross_val_only"}
-    enable_remote_logging: bool
-        Flag to enable logging of model building metadata to a remote backend.  By
-        default remote logging is disabled and the MlFlow client logs to a local
-        `mlruns/` directory.
     """
+    if model_parameter and isinstance(machine["model"], str):
+        model_parameter = dict(model_parameter)
+        machine["model"] = expand_model(machine["model"], model_parameter)
 
-    # Set default data provider for data config
-    # TODO: This is for backwards compatibility, as the `data_provider` param should
-    # TODO: be provided in the `data_config` itself
-    if data_provider is not None:
-        data_config["data_provider"] = data_provider
+    machine = Machine.from_config(machine, project_name=machine["project_name"])
```

Comment:
I find it a bit confusing to read code (in general also, yes LOL) that reuses variables, in particular when they change type and "meaning". Something like machine_config or machine_dict perhaps? It's just when reading the code, jumping a bit up and down, when it has changed to something else in-the-middle it confuses me.

Also, type hints would be cool-o-yeah. Since you are changing it anyways, and already have it in the docstring. Or is there some reason for omitting it I'm missing here perhaps?

Reply:
Ya, I agree. I fix, I fix!